### PR TITLE
[FLINK-28522][tests][JUnit5 migration] flink-sequence-file

### PR DESCRIPTION
**Pull request by RyanSkraba**
## What is the purpose of the change

Update the `flink-formats/flink-sequence-file` module to AssertJ and JUnit 5 following the [JUnit 5 Migration Guide](https://docs.google.com/document/d/1514Wa_aNB9bJUen4xm5uiuXOooOJTtXqS_Jqk9KJitU/edit)

## Brief change log

* Removed dependences on JUnit 4, JUnit 5 Assertions and Hamcrest where possible.

## Verifying this change

This change is a code cleanup without any test coverage.

I verified that there were 4 tests run before and after the change.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive):no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by RyanSkraba** [comment]
@flinkbot run azure

**Review by XComp — COMMENTED** [review]
LGTM :+1: Again, some minor cosmetic comments. I'm curious what you think. Please rebase the branch to make it ready to being merged.

File: flink-formats/flink-sequence-file/src/test/java/org/apache/flink/formats/sequencefile/SerializableHadoopConfigurationTest.java [hunk 1]
```diff
@@ -19,70 +19,51 @@
...
-                        .appendValue(TEST_KEY)
-                        .appendText(" and value=")
-                        .appendValue(TEST_VALUE);
-            }
-        };
+        Assertions.<Configuration>assertThat(deserializableConfigUnderTest.get())
+                .matches(
+                        actualConfig -> {
+                            final String value = actualConfig.get(TEST_KEY);
+                            return actualConfig != configuration
```
[the remark this answers is not in the document]
Comment (RyanSkraba) [reply flink-formats/flink-sequence-file/src/test/java/org/apache/flink/formats/sequencefile/SerializableHadoopConfigurationTest.java]: This needed to be rewritten in order to compile.   I squashed it into the single commit.

**Comment by XComp** [comment]
`KafkaShuffleExactlyOnceITCase` failed due to FLINK-24119. I added the pipeline run to the ticket already. I'm going to investigate the `PulsarSourceUnorderedE2ECase` test failure (see [build failure](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=42325&view=logs&j=af184cdd-c6d8-5084-0b69-7e9c67b35f7a&t=160c9ae5-96fd-516e-1c91-deb81f59292a&l=14720))

**Review by XComp — CHANGES_REQUESTED** [review]
We seem to use the description logic in the wrong way. Please have a look.

**Comment by XComp** [comment]
I created FLINK-29755 to cover the `PulsarSourceUnorderedE2ECase.testSavepoint` test failure in the [failed build](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=42325&view=logs&j=af184cdd-c6d8-5084-0b69-7e9c67b35f7a&t=160c9ae5-96fd-516e-1c91-deb81f59292a&l=14380)

File: flink-formats/flink-sequence-file/src/test/java/org/apache/flink/formats/sequencefile/SerializableHadoopConfigurationTest.java [hunk 2]
```diff
@@ -19,70 +19,54 @@
...
+                        actualConfig -> {
+                            final String value = actualConfig.get(TEST_KEY);
+                            return actualConfig != serializableConfigUnderTest.get()
+                                    && value != null
+                                    && serializableConfigUnderTest
+                                            .get()
+                                            .get(TEST_KEY)
+                                            .equals(value);
+                        })
+                .describedAs(
```
[the remark this answers is not in the document]
Comment (RyanSkraba) [reply flink-formats/flink-sequence-file/src/test/java/org/apache/flink/formats/sequencefile/SerializableHadoopConfigurationTest.java:67]: Yes, it seems that the `describedAs` needs to come before any asserts!

In this case, I changed the logic quite a bit from the original to improve the descriptive error messages.  The new logic would print the error message:

```
[ERROR] Failures: 
[ERROR]   SerializableHadoopConfigurationTest.customPropertiesSurviveSerializationDeserialization:60 
[a Hadoop Configuration with property: key=test-key and value=test-value] (1 failure)
-- failure 1 --
expected: "test-value"
 but was: "faked-for-error-message"
```

**Review by XComp — APPROVED** [review]

**Comment by XComp** [comment]
I accidentally cancelled the squash-merge but the corresponding commit 276bb778c7ea4555370c6155af5128ea7f7cbe6c ended up on `master` as planned. Therefore, I'm manually closing this PR. The PR is considered `MERGED`.

**Pull request closed without merging**